### ISSUE TITLE
package:js based JS interop is broken with DDK, likely missing annotations in kernel outlines

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/sdk#36912]
See original issue https://github.com/dart-lang/build/issues/2260 with repro package here https://github.com/valakis/test_repo.

Typically the dev compiler should inline a call directly at the call site of the `bridge` method like this `dart.global.bridge()`, but instead we are seeing it try to call the `bridge` method on the `dart_interop` module directly, which doesn't exist (the module is effectively empty, as intended).

It looks for the `@JS` annotations to make this decision, so the most likely scenario is that the annotation is missing from the outline for that module. I attached the outline file if somebody wants to confirm that, but the output from `pkg/kernel/bin/dump.dart` looks like this:

```
main = <No Member>;
library bridge from "org-dartlang-app:///test/bridge_interop.dart" as bri {

  import "package:js/js.dart";

  external static abstract method bridge() → core::String;
}
```

[bridge_interop.ddc.dill.zip](https://github.com/dart-lang/sdk/files/3162773/bridge_interop.ddc.dill.zip).

**Comment by jakemac53** [comment]
cc @kmillikin @vsmenon @natebosch @jacob314

**Comment by jakemac53** [comment]
I also wonder if in general it isn't preferable to omit these annotations, and we could instead inject some extra metadata using a kernel transformer based on this annotation, just for the DevCompiler target?

**Comment by vsmenon** [comment]
I filed https://github.com/dart-lang/sdk/issues/36944 to separate out the front end issue.

**Comment by evanweible-wf** [comment]
@jakemac53 I tried the workaround from https://github.com/dart-lang/build/issues/2260 in order to get a version of react-dart that works on dart 2.3.0 + build_web_compilers 2.0.0, but unfortunately with certain strongly typed functions/methods, we hit the following DDC issue:

https://github.com/dart-lang/sdk/issues/36114

I don't think we have a way to workaround that issue without dropping the types, so we are unable to upgrade to build_web_compilers 2.0.0 even with the JS interop workaround.

**Comment by vsmenon** [comment]
Update on this: @kmillikin expects a fix to the upstream #36944 soon.  That should fix this cleanly.  Keeping this open to validate.

**Comment by vsmenon** [comment]
@kmillikin 's pending CL appears to fix this:

https://dart-review.googlesource.com/c/sdk/+/103680

**Comment by dgrove** [comment]
Note that fix for #36944 had to be reverted due to impact on Flutter benchmarks.

**Comment by aadilmaan** [comment]
Considering this as a closed/dupe of https://github.com/dart-lang/sdk/issues/36944.
